File: langs/javascript/collatz/MaxSequence.js
#!/usr/bin/env node

'use strict';

const NUMBER = 500000;

function collatz(x)
{
    var len = 0;

    while (x > 1) {
        if (x % 2 == 0) {
            x = x / 2;
        } else {
            x = 3 * x + 1;
        }

        len++;
    }

    return len;
}

function findMaxCollatz(to) {
    var result = [1, 1];

    for (var number = 1; number <= to; number++) {
        var len = collatz(number);

        if (len > result[1]) {
            result = [number, len];
        }
    }

    return result;
}

(async function() {
    const startTimeMs = new Date().getTime();
    
    console.log(findMaxCollatz(NUMBER));

    const endTimeMs = new Date().getTime();
    const durationMs = endTimeMs - startTimeMs;
    console.log(`Execution time: ${durationMs}ms`);
})();

File: langs/ruby/collatz/MaxSequence.rb
NUMBER = 500_000

def collats(x)
    len = 0

    while x > 1 do
        if x % 2 == 0
            x = x / 2
        else
            x = 3 * x + 1
        end

        len += 1
    end

    return len
end

def findMaxCollatz(to)
    result = [1, 1]

    for number in 1..to
        len = collats(number)

        if len > result[1]
            result = [number, len]
        end
    end

    return result
end

startTimeMs = Time.now.to_f * 1000

puts findMaxCollatz(NUMBER)

endTimeMs = Time.now.to_f * 1000
durationMs = (endTimeMs - startTimeMs).to_i

puts "Execution time: #{durationMs}ms"